test/integration/controller/core/workload_controller_test.go

```diff
@@ -209,4 +215,88 @@ var _ = ginkgo.Describe("Workload controller", func() {
+			priorityClass = testing.MakePriorityClass("priority-class").PriorityValue(100).Obj()
+			gomega.Expect(k8sClient.Create(ctx, workloadPriorityClass)).To(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, priorityClass)).To(gomega.Succeed())
+
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+			gomega.Expect(k8sClient.Delete(ctx, workloadPriorityClass)).To(gomega.Succeed())
+			gomega.Expect(k8sClient.Delete(ctx, priorityClass)).To(gomega.Succeed())
+		})
+		ginkgo.It("case of WorkloadPriorityClass", func() {
+			wl = testing.MakeWorkload("wl", ns.Name).Queue("lq").Request(corev1.ResourceCPU, "1").
+				PriorityClass("workload-priority-class").PriorityClassSource(constants.WorkloadPriorityClassSource).Priority(200).Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+			gomega.Eventually(func() []metav1.Condition {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), &updatedQueueWorkload)).To(gomega.Succeed())
+				return updatedQueueWorkload.Status.Conditions
+			}, util.Timeout, util.Interval).ShouldNot(gomega.BeNil())
+			initialPriority := int32(200)
+			gomega.Expect(updatedQueueWorkload.Spec.Priority).To(gomega.BeComparableTo(&initialPriority))
+			updatedPriority := int32(150)
+			updatedQueueWorkload.Spec.Priority = &updatedPriority
+			gomega.Expect(k8sClient.Update(ctx, &updatedQueueWorkload)).To(gomega.Succeed())
+			gomega.Eventually(func() int32 {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(&updatedQueueWorkload), &finalQueueWorkload)).To(gomega.Succeed())
+				return *finalQueueWorkload.Spec.Priority
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(updatedPriority))
+		})
+		ginkgo.It("case of PriorityClass", func() {
+			wl = testing.MakeWorkload("wl", ns.Name).Queue("lq").Request(corev1.ResourceCPU, "1").
+				PriorityClass("priority-class").PriorityClassSource(constants.PodPriorityClassSource).Priority(100).Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+			gomega.Eventually(func() []metav1.Condition {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), &updatedQueueWorkload)).To(gomega.Succeed())
+				return updatedQueueWorkload.Status.Conditions
+			}, util.Timeout, util.Interval).ShouldNot(gomega.BeNil())
+			initialPriority := int32(100)
+			gomega.Expect(updatedQueueWorkload.Spec.Priority).To(gomega.BeComparableTo(&initialPriority))
+			updatedPriority := int32(50)
+			updatedQueueWorkload.Spec.Priority = &updatedPriority
+			gomega.Expect(k8sClient.Update(ctx, &updatedQueueWorkload)).To(gomega.Succeed())
+			gomega.Eventually(func() int32 {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(&updatedQueueWorkload), &finalQueueWorkload)).To(gomega.Succeed())
+				return *finalQueueWorkload.Spec.Priority
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(updatedPriority))
+		})
+	})
```

Comment:
It looks like these tests verify workload webhooks. So can we move these to https://github.com/kubernetes-sigs/kueue/blob/025384a6d4d118538233d1603214b32a2a30b4bd/test/integration/webhook/workload_test.go?

Reply:
I see. This test confirms that workload webhooks wouldn't stop updating the priority values of workload. So it might be the test case for webhook instead of controller.